### DISCUSSION TITLE
How to test a simple route like blog/:slug ?

### DISCUSSION THREAD

**Discussion by sarthaksavvy** [vercel/next.js#12703] [Help]
I have a simple nextjs app where I need to write tests to validate that blog/:slug will work on both client-side and on the server side

I am using jest for testing

**Comment by leo-petrucci** [comment]
Asked this a couple of hours back but didn't get a response. Following this for interest.

**Comment by rafaelalmeidatk** [comment]
I am not sure if you can do it with Jest, but you can test the everything together using Cypress, since it will test the app like an user in the browser: https://www.cypress.io

**Comment by craigmulligan** [comment]
I've written a short [post](https://hobochild.com/posts/tests-next-js) on how I test next apps, it may be helpful. Let me know if you have any questions.
